docs/development/go/file-io/index.md

```diff
@@ -0,0 +1,720 @@
+        if err == io.EOF {
+            break
+        } else if err != nil {
+            fmt.Printf("error reading file %s", err)
+            break
+        }
+        fmt.Print(line)
+    }
+    return nil
+}
+
+func main() {
+    flag.Parse()
+    if len(flag.Args()) == 0 {
+        fmt.Printf("usage: lByL <file1> [<file2> ...]\n")
+        return
+    }
+
+    for _, file := range flag.Args() {
+        err := lineByLine(file)
+        if err != nil {
+            fmt.Println(err)
+        }
+    }
+}
+{{< /file >}}
+
+The functionality of the program can be found in the `lineByLine()` function. After making sure that you can open the given filename, you create a new reader using `bufio.NewReader()`. Then, you use that reader with `bufio.ReadString()` in order to read the input file line by line. The trick is done by the parameter of `bufio.ReadString()`, which is a character that tells `bufio.ReadString()` to keep reading until that character is found. Constantly calling `bufio.ReadString()` when that parameter is the *newline character* results in reading the input file line by line. The `for` loop in the `main()` function is for being able to process multiple command line arguments.
+
+Executing `lByL.go` will generate the following kind of output:
+
+    go run lByL.go /tmp/data.txt
+{{< output >}}
+1 2
+One Two
+
+Three
+{{< /output >}}
+
+### Reading a Text File Word by Word
+
+The presented code shows how you can read a text file word by word.
+
+{{< file "./wByW.go" go >}}
+package main
+
+import (
+    "bufio"
+    "flag"
+    "fmt"
+    "io"
+    "os"
+    "regexp"
+)
+
+func wordByWord(file string) error {
+    var err error
+    fd, err := os.Open(file)
+    if err != nil {
+        return err
+    }
+    defer fd.Close()
+
+    reader := bufio.NewReader(fd)
+    for {
+        line, err := reader.ReadString('\n')
+        if err == io.EOF {
+            break
+        } else if err != nil {
+            fmt.Printf("error reading file %s", err)
+            return err
+        }
+
+        re := regexp.MustCompile("[^\\s]+")
+        words := re.FindAllString(line, -1)
+        for i := 0; i < len(words); i++ {
+            fmt.Println(words[i])
+        }
+    }
+    return nil
+}
+
+func main() {
+    flag.Parse()
+    if len(flag.Args()) == 0 {
+        fmt.Printf("usage: wByW <file1> [<file2> ...]\n")
+        return
+    }
+
+    for _, file := range flag.Args() {
+        err := wordByWord(file)
+        if err != nil {
+            fmt.Println(err)
+        }
+    }
+}
+{{< /file >}}
+
+The functionality of the program can be found in the `wordByWord()` function. Initially the text file is read line by line. Then a regular expression, which is stored in the `r` variable, is used for finding out the words in the current line and store them in the `words` variable. After that, a `for` loop is used for iterating over the contents of `words` and print them on the screen before continuing with the next line of the input file.
```

Comment:
I'm not able to see where the `r` variable exists in this code sample. Would the variable be called `re` instead?

Reply:
It should be `re` - you are correct.